### PR TITLE
Issue #12079: Github action Release perform

### LINKED ISSUE #12079: Github action "Release perform"
https://github.com/checkstyle/contribution/blob/master/releasenotes-builder/diagrams-manual-trigger.md

we need to make action to do `run Github action "Push perform '10.3.1'"` (action takes human provided version as input, just to make sure it can not be executed unintentionally)
if user provider input is not matching pom - fail build

so we simply make script that do checkout to required tag and do `mvn:release perform`.
this phase suppose to not push anything to github,
but it should able to deploy jars to maven central.

For testing purposes in your test repo you can do release to some folder, you need to configure release plugin to do so.

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Fixes #12079

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
Need some time to test it. Facing some errors, resolving them and will get back to the generation soon

File: .ci/maven-perform.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+echo CURRENT_VERSION="$CURRENT_VERSION"
+
+if [ "$TARGET_VERSION" != "$CURRENT_VERSION" ]; then
+  echo "[ERROR] Target Version and Current Version doesn't match."
+  exit 1;
+fi
+
+SKIP_CHECKSTYLE="-Dcheckstyle.ant.skip=true -Dcheckstyle.skip=true"
+
+git fetch --all --tags
```
Comment (romani) [inline .ci/maven-perform.sh:24]: why clone in action is not enough ?

File: .ci/maven-perform.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [[ -z $1 ]]; then
+  echo "latest version is not set"
```
Comment (romani) [inline .ci/maven-perform.sh:6]: remove latest

**Review by romani — CHANGES_REQUESTED** [review]
show me your maven project that you use to test this action.

items:

**Comment by Rahulkhinchi03** [comment]
This is my project: https://github.com/Rahulkhinchi03/checkstyle-test

**Comment by romani** [comment]
Remove this file https://github.com/Rahulkhinchi03/checkstyle-test/blob/main/release.properties it should not be in git

File: .ci/maven-perform.sh (hunk 2 above)
Comment (Rahulkhinchi03) [reply .ci/maven-perform.sh]: Changes Done

File: .ci/maven-perform.sh (hunk 1 above)
Comment (Rahulkhinchi03) [reply .ci/maven-perform.sh]: Changes Done

File: .github/workflows/maven_perform.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        description: 'New Version'
+        required: true
+jobs:
+  perform:
+    name: Push perform {{ github.event.inputs.version }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the latest code
+        uses: actions/checkout@v2
+        with:
```
Comment (romani) [inline .github/workflows/maven_perform.yml:19]: Checkout is not enough, no tags

File: .ci/maven-perform.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+if [ "$TARGET_VERSION" != "$CURRENT_VERSION" ]; then
+  echo "[ERROR] Target Version and Current Version doesn't match."
+  exit 1;
+fi
+
+SKIP_CHECKSTYLE="-Dcheckstyle.ant.skip=true -Dcheckstyle.skip=true"
+
+git checkout "checkstyle-$TARGET_VERSION"
+echo "Deployment of jars to maven central (release:perform) ..."
+mvn -e --no-transfer-progress release:perform -Darguments="$SKIP_CHECKSTYLE"
```
Comment (romani) [inline .ci/maven-perform.sh:27]: Please prove by testing that this command can run successfully without release:prepare on same host/filesystem.

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Pull request closed without merging**